src/components/SavedRecipes.tsx

```diff
@@ -1,13 +1,18 @@
-import React, { useState, useEffect } from "react";
+import React, { useState, useEffect, useCallback } from "react";
 import { Center, Container, Flex, Grid, GridItem, Text } from "@chakra-ui/react";
 import { SavedRecipe } from "../utils/types";
 import { getRecipe } from "../utils/fetchData";
 import SavedRecipesList from "./SavedRecipesList";
 import CategoriesList from "./CategoriesList";
+import { useSearchParams } from "react-router-dom";
 
 const SavedRecipes = () => {
   const [recipes, setRecipes] = useState([]);
-  const [filteredRecipes, setFilteredRecipes] = useState(recipes);
+  const [filteredRecipes, setFilteredRecipes] = useState([]);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [show, setShow] = useState(true);
+
+  let params = searchParams.get("search");
```

Comment:
I know this is the naming convention that react-router-dom uses, but I would do this instead.
```
const searchQueryParam = searchParams.get("search");
const searchQueryParamParsed = searchQueryParam.toLowerCase();
```

Reply:
Done! 🤝